sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RetryContext.java

```diff
@@ -7,43 +7,52 @@
 
 import java.time.Duration;
 import java.time.Instant;
+import java.util.ArrayList;
 import java.util.List;
 
 public class RetryContext {
-
-    @JsonIgnore
-    public List<int[]> directRetrySpecificStatusAndSubStatusCodes;
-    @JsonIgnore
-    public List<int[]> genericRetrySpecificStatusAndSubStatusCodes;
-
     @JsonIgnore
-    public Instant retryStartTime;
+    public volatile Instant retryStartTime;
     @JsonIgnore
-    public Instant retryEndTime;
+    public volatile Instant retryEndTime;
 
-    public int retryCount;
+    private volatile List<int[]> statusAndSubStatusCodes;
```

Comment:
Suggestions:
- Using a concurrent collection would be safer if multiple threads would be updating this collection
- Using a `Pair<>` for status and substatus codes would be more readable than unbounded `int[]`

Reply:
Yes concurrent list is good here, done. On status and substatus, dont want to touch the existing data structure. Also int[] is only updated during initializations it should be fine.